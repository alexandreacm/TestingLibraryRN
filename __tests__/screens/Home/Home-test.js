import React from 'react';
import App from '../../../App';

import renderer, { create } from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';

describe('HomeScreen', () => {
  const tree = renderer.create(<App />).toJSON();

  test('Should run snapshot test', () => {
    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly component', () => {
    render(<App />);
  });

  test('Should render correctly ap with react-test-renderer', () => {
    create(<App />);
  });

  test('Should Create an item', () => {
    const { getByText, getByPlaceholderText } = render(<App />);

    const input = getByPlaceholderText('Write something');
    const button = getByText('+');

    const createdItemText = 'first todo';

    fireEvent.changeText(input, createdItemText);
    fireEvent.press(button);

    const createdItem = getByText(createdItemText);
    expect(createdItem).not.toBeNull();
    // expect(createdItem).not.toBe(null);
  });

  test('Should Create an item and test an item with RegEx', () => {
    const { getByText, getAllByText, getByPlaceholderText } = render(<App />);

    const input = getByPlaceholderText('Write something');
    const button = getByText('+');

    const createdItemText = 'first todo';

    fireEvent.changeText(input, createdItemText);
    fireEvent.press(button);

    const createdItem = getAllByText(/first todo/i);
    expect(createdItem.length).toBe(1);
  });

  test('Should Create an item and check with toBeDefined', () => {
    const { getByText, getByPlaceholderText } = render(<App />);

    const input = getByPlaceholderText('Write something');
    const button = getByText('+');

    const createdItemText = 'first todo';

    fireEvent.changeText(input, createdItemText);
    fireEvent.press(button);

    const createdItem = getByText(createdItemText);
    expect(createdItem).toBeDefined();
  });

  test('Should Create an item through getByTestID', () => {
    const { getByText, getByTestId, getByPlaceholderText } = render(<App />);

    const input = getByPlaceholderText('Write something');
    const button = getByTestId('btnTodo');

    const createdItemText = 'first todo';

    fireEvent.changeText(input, createdItemText);
    fireEvent.press(button);

    const createdItem = getByText(createdItemText);
    //expect(createdItem).not.toBe(null);
    expect(createdItem).not.toBeNull();
  });

  test('Should create multiple items', () => {
    const { getByText, getByTestId, getByPlaceholderText } = render(<App />);

    const input = getByPlaceholderText('Write something');
    const button = getByTestId('btnTodo');

    const createdItemText = 'first todo';
    const createdItemTextSecond = 'second todo';

    fireEvent.changeText(input, createdItemText);
    fireEvent.press(button);

    fireEvent.changeText(input, createdItemTextSecond);
    fireEvent.press(button);

    const createdItem = getByText(createdItemText);
    const createdItemSecond = getByText(createdItemTextSecond);

    expect(createdItem).not.toBeNull();
    expect(createdItemSecond).not.toBeNull();
  });

  test('Should delete an item', () => {
    const { getByText, queryByText, getByTestId, getByPlaceholderText } =
      render(<App />);

    const input = getByPlaceholderText('Write something');
    const button = getByTestId('btnTodo');

    const createdItemText = 'first todo';

    fireEvent.changeText(input, createdItemText);
    fireEvent.press(button);

    const buttonDelete = getByText('X');
    fireEvent.press(buttonDelete);

    const deleteItem = queryByText(createdItemText);

    expect(deleteItem).toBeNull();
    //expect(deleteItem).toBe(null);
  });

  test('Should show an error when trying to create an item without text', () => {
    const { getByText, getByTestId } = render(<App />);

    const button = getByTestId('btnTodo');
    fireEvent.press(button);

    // const createdItemText = 'Please insert a valid text';
    const errorMessage = getByText('Please insert a valid text');

    // expect(errorMessage).not.toBe(null);
    expect(errorMessage).not.toBeNull();
  });

  test('Should remove the error message after creating a valid item', () => {
    const { queryByText, getByTestId, getByPlaceholderText } = render(<App />);

    const createdItem = 'first todo';
    const createdItemErrorMessage = 'Please insert a valid text';

    const input = getByPlaceholderText('Write something');
    const button = getByTestId('btnTodo');

    fireEvent.press(button);

    fireEvent.changeText(input, createdItem);
    fireEvent.press(button);

    const errorMessage = queryByText(createdItemErrorMessage);

    expect(errorMessage).toBeNull();
  });
});
