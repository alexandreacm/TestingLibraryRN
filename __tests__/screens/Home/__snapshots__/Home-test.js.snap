// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Should run snapshot test 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 20,
      "marginTop": 60,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Write something"
        placeholderTextColor="#CDCDCD"
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#000",
            "borderRadius": 10,
            "fontSize": 18,
            "height": 50,
            "width": "85%",
          }
        }
        testID="txtTodo"
        value={null}
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#503FDE",
            "borderRadius": 10,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "15%",
          }
        }
        testID="btnTodo"
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontSize": 30,
              "textAlign": "center",
            }
          }
          testID="txtTodo"
        >
          +
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 10,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FF0000",
            "fontSize": 16,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
          "marginTop": 30,
        }
      }
    >
      <RCTScrollView
        data={Array []}
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        testID="flatTodo"
        viewabilityConfigCallbackPairs={Array []}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</View>
`;
